### PR TITLE
Add plone 5 profile for ftw.addressblock geo

### LINKED ISSUE #26: Missing plone 5 profile for geo extra
The profile `ftw/addressblock/geo/profiles/default` contains a `jsregistry.xml` file, which should be ported to plone 5's `registry.xml`. 

The `authenticated` attribute can be achieved by using an expression like this in the bundle: 
```xml
<value key="expression">python: member is not None</value>
```

Attention here because only one of the 2 defined javascript is for authenticated users only not both.

### PR DESCRIPTION
**Pull request by busykoala**
#26 
https://github.com/4teamwork/izug.organisation/issues/1622

PLONE5: Except of https://github.com/4teamwork/ftw.addressblock/issues/31 the plone5 is now ready when `plonetheme.blueberry` gets installed with the package.

PLONE4: This is pretty broken right now. There are several things to look into. It might be a good idea to not continue both version on the same branch.

### CODE REVIEW

File: ftw/addressblock/geo/profiles/default_plone5/registry.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0"?>
+<registry>
+
+    <records interface="collective.geo.settings.interfaces.IGeoFeatureStyle">
+        <value key="map_viewlet_position">plone.belowcontentbody</value>
+    </records>
+
+    <records prefix="plone.resources/ftw-addressblock-js-authenticated"
+             interface='Products.CMFPlone.interfaces.IResourceRegistry'>
+        <value key="expression">python: member is not None</value>
```
Comment (Nachtalb) [inline ftw/addressblock/geo/profiles/default_plone5/registry.xml]: Have you tested this? 
The `expression` is a field on the bundle not on the js itself.
https://github.com/plone/Products.CMFPlone/blob/9ae305beb3abd55cf47e05425e19f201768c804a/Products/CMFPlone/interfaces/resources.py#L63-L68

Comment (busykoala) [reply ftw/addressblock/geo/profiles/default_plone5/registry.xml]: No, I did not. I was relying on your comment how to do it in issue #26.
Also I would expect the tests to fail if is relevant to the security.

Comment (Nachtalb) [reply ftw/addressblock/geo/profiles/default_plone5/registry.xml]: Sorry, my issue should have been more clear.

Comment (busykoala) [reply ftw/addressblock/geo/profiles/default_plone5/registry.xml]: I think I fixed this now.

**Comment by busykoala** [comment]
❗️ We should not yet merge this. There are still issues with the map. ❗️

**Comment by Nachtalb** [comment]
https://github.com/4teamwork/ftw.simplelayout/pull/535 fixes the problem with the map not loading in the mapblock form. Due to changes in the HTML structure of tabs in forms, the mapblock did not initialise the map when clicking on the `Coordinates` tab.

**Comment by busykoala** [comment]
Now having added `ftw.addressblock [geo]` the behaviour acts strange. When adding into search input for searching locations, on enter/send the tab changes and the map does not update accordingly.
Also if an address is entered only open street map seems to work.

**Comment by Nachtalb** [comment]
`mapblock` extra from `ftw.simplelayout` was missing and the click event was not caught ==> https://github.com/4teamwork/ftw.simplelayout/pull/539

**Comment by busykoala** [comment]
@Nachtalb, is this fixed and can be reviewed, or do we still have a problem here? I can't remember wether your last commit actually fixed the block or if we then found another issue with it...?

**Comment by busykoala** [comment]
@maethu, it is probably not very useful reading all the comments in-between since much changed in the meanwhile. It would be great if you could look at the TODOs in the first comment at the top.

**Comment by maethu** [comment]
Regarding scrolling:
It's a theme issue, simplelayout and the plone 5 theme do not work perfectly out of the box.
It works for example fine with plonetheme.blueberry: see -->
Screen Shot 2019-11-14 at 20 37 05

**Comment by maethu** [comment]
Regarding the missing icon:
Weird if I enable dev mode for openlayers bundle it works...

I feel we can investigate that later. I can't figure it out right now.

**Comment by busykoala** [comment]
Because sources in an external dependency can not be loaded I overwrote the image for this block. Also it is using symbols instead of images now.  
The settings are done within the theming resources.

image

image

**Comment by busykoala** [comment]
I removed the bug with the overlay and opened https://github.com/4teamwork/ftw.addressblock/issues/31 to look at that separately.

**Comment by maethu** [comment]
Here we go, latest gossip regarding ftw.addressblock:

1. if the map is no loaded via tab, eg. mapblock - everything works fine with plone4 and the ftw.addressblock master / ftw.simplelayout master
2. The configuration from PROD zg.ch (ftw.simplelayout 1.23.9 and ftw.addressblock 1.0.4, which was considered a working combo, does only work if the map is NOT a google map, but a OpenlayersMap, or the ZugMap. If google map is active -> same Problem as with the latest version

The conclusion so far is, that the current Openlayers version (2.14 dev) is probably not compatible anymore with the google api in all circumstances. 

I'm confused though about the fact, that it works off the tab :thinking:

**Comment by maethu** [comment]
`You have included the Google Maps JavaScript API multiple times on this page. This may cause unexpected errors.` lead to this problem `InvalidValueError: setCenter: not a LatLng or LatLngLiteral with finite coordinates: in property lat: not a number` due an update in the google api v3 script, which we include directly from google.

**Comment by maethu** [comment]
@busykoala GREAT SUCCESS

With..
- https://github.com/4teamwork/ftw.simplelayout/pull/564
- https://github.com/4teamwork/ftw.addressblock/pull/32
- https://github.com/4teamwork/ftw.colorbox/pull/19

I was able make addressblock work in plone4 and plone5
- Add addressblock search for address in overlay 
- Add addressblock set point manually
- Edit AddressBlock WITH overlay (set address manually and search for an address)
- Edit AddressBlock WITHOUT overlay (set address manually and search for an address)

Hint: Keep in mind, an address is need for addressblock, so if you add an addressblock and leave the address empty the coordinates are not saved. Seems to be a restriction with ftw.geo. --> This problem exists, since forever and we don't have to investigate it now.

IMHO it's no working as expected.

**Comment by busykoala** [comment]
In https://github.com/4teamwork/ftw.addressblock/pull/33 I tried to rebase and autocheckout the pkgs ftw.simplelayout and ftw.colorbox (to get the master with changes required).
I did not succeed though. Please do not merge before looking at it again ;-)

**Comment by busykoala** [comment]
I close this branch in favour of #33

**Pull request closed without merging**